Add Windows build to CI

jakobnissen opened the pull request:
Let's see if this works

jakobnissen committed:
Add Windows build to CI

apcamargo committed:
Rename windows build CI

apcamargo commented:
Thanks for the PR! The maturin step is failing before the build starts. I'm on a tight schedule right now, but I can take a look at this early next week.

jakobnissen committed:
Try making Windows work

jakobnissen committed:
Another try

jakobnissen committed:
Another try

jakobnissen committed:
Final try

jakobnissen committed:
Update dependencies

Update all dependencies to latest versions. This might be necessary to support
Windows.

jakobnissen committed:
Try again

jakobnissen committed:
Merge branch 'main' into windows

jakobnissen commented:
Okay I'm stumped - I still can't get `rust-htslib` to compile. Maybe the solution is to have openssl installed on the Windows machine compiling rust-htslib. The rust-htslib README.md implies that it CAN be installed on Windows.

I've temporarily outcommented all CI except Python 3.10 on Windows to not burn too much compute.
